pkg/controller/worker/machines_test.go

```diff
@@ -640,6 +641,48 @@ var _ = Describe("Machines", func() {
 					Expect(err).NotTo(HaveOccurred())
 					Expect(result).To(Equal(machineDeployments))
 				})
+
+				It("should delete the any old AWSMachineClasses", func() {
+					// mockAWSMachineClassList is the mocking object used to track the
+					// list of AWSMachineClass objects
+					mockAWSMachineClassList := machinev1alpha1.AWSMachineClassList{
+						Items: []machinev1alpha1.AWSMachineClass{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-0",
+								},
+							},
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-1",
+								},
+							},
+						},
+					}
+
+					// Initialize mocking functions
+					expectGetSecretCallToWork(c, awsAccessKeyID, awsSecretAccessKey)
+					expectDeleteAllCallToWork(c, &mockAWSMachineClassList)
+
+					workerCloudRouter := w
```

Comment:
workerCloudRouter sounds like a leftover from provider-gcp

```suggestion
					workerDelegate, _ := NewWorkerDelegate(common.NewClientContext(c, scheme, decoder), chartApplier, "", w, cluster)
```

Reply:
Made the suggested change here - https://github.com/gardener/gardener-extension-provider-aws/pull/212/files#diff-12a1fa4ba4f4250731a53989c476c9407e832ccecae17ad3fc1b2eee100ad7c3R667.